react/features/web-hid/middleware.ts

```diff
@@ -0,0 +1,161 @@
+        updateDeviceListener = _updateDeviceListener;
+
+        hidManager.listenToConnectedHid();
+        hidManager.addEventListener(EVENT_TYPE.INIT_DEVICE, initDeviceListener);
+        hidManager.addEventListener(EVENT_TYPE.UPDATE_DEVICE, updateDeviceListener);
+
+        break;
+    }
+    case APP_WILL_UNMOUNT: {
+        const hidManager = getWebHidInstance();
+
+        if (!isDeviceHidSupported()) {
+            break;
+        }
+
+        if (typeof initDeviceListener === 'function') {
+            hidManager.removeEventListener(EVENT_TYPE.INIT_DEVICE, initDeviceListener);
+        }
+        if (typeof updateDeviceListener === 'function') {
+            hidManager.removeEventListener(EVENT_TYPE.UPDATE_DEVICE, updateDeviceListener);
+        }
+
+        hidManager.close();
+
+        break;
+    }
+    case CLOSE_HID_DEVICE: {
+        if (!isDeviceHidSupported()) {
+            logger.info('HID device not supported');
+
+            break;
+        }
+
+        const hidManager = getWebHidInstance();
+
+        // cleanup event handlers when hid device is removed from Settings.
+        if (typeof initDeviceListener === 'function') {
+            hidManager.removeEventListener(EVENT_TYPE.INIT_DEVICE, initDeviceListener);
+        }
+        if (typeof updateDeviceListener === 'function') {
+            hidManager.removeEventListener(EVENT_TYPE.UPDATE_DEVICE, updateDeviceListener);
+        }
+
+        hidManager.close();
+        dispatch(muteLocal(false, MEDIA_TYPE.AUDIO));
+
+        break;
+    }
+    case REQUEST_HID_DEVICE: {
+        if (!isDeviceHidSupported()) {
+            logger.info('HID device not supported');
+
+            break;
+        }
+
+        const hidManager = getWebHidInstance();
+
+        await hidManager.requestHidDevices();
+
+        if (!hidManager.availableDevices || !hidManager.availableDevices.length) {
+            logger.info('HID device not available');
+            break;
+        }
+
+        const _initDeviceListener = (e: any) => dispatch(initDeviceInfo(e.detail.deviceInfo));
+        const _updateDeviceListener = (e: any) => {
+            dispatch(updateDeviceInfo(e.detail.deviceInfo));
+
+            if (e.detail?.actionResult?.eventName === ACTION_HOOK_TYPE_NAME.MUTE_SWITCH_ON) {
+                dispatch(muteLocal(true, MEDIA_TYPE.AUDIO));
+            } else if (e.detail?.actionResult?.eventName === ACTION_HOOK_TYPE_NAME.MUTE_SWITCH_OFF) {
+                dispatch(muteLocal(false, MEDIA_TYPE.AUDIO));
+            }
+        };
+
+        initDeviceListener = _initDeviceListener;
+        updateDeviceListener = _updateDeviceListener;
+
+        hidManager.addEventListener(EVENT_TYPE.INIT_DEVICE, initDeviceListener);
+        hidManager.addEventListener(EVENT_TYPE.UPDATE_DEVICE, updateDeviceListener);
+        await hidManager.listenToConnectedHid();
+
+        const localTracks = getLocalTracks(store.getState()['features/base/tracks']);
```

Comment:
Check the helpers in base/media instead of checking the tracks.

Reply:
done